docs/mp/guides/04_health.adoc

```diff
@@ -291,10 +277,121 @@ curl -v http://localhost:8080/health/ready
 ----
 <1> The HTTP status is `200` indicating that the application is ready.
 
+=== Custom Startup Health Checks
+
+You can add a startup check to indicate whether or not the application has initialized to the point that the other health checks make sense.
+In this example, the server will wait eight seconds before it declares itself started.
+
+[source,java]
+.Create a new `GreetStartedCheck` class with the following content:
+----
+
+package io.helidon.examples.quickstart.mp;
+
+import java.time.Duration; // <1>
+import java.util.concurrent.atomic.AtomicLong;
+import jakarta.enterprise.context.ApplicationScoped;
+
+import jakarta.enterprise.context.Initialized;
+import jakarta.enterprise.event.Observes;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.Started;
+
+@Started // <2>
+@ApplicationScoped
+public class GreetStartedCheck implements HealthCheck {
+  private AtomicLong readyTime = new AtomicLong(0);
+
+
+  @Override
+  public HealthCheckResponse call() {
+    return HealthCheckResponse.named("StartedCheck")  // <3>
+        .status(isReady())
+        .withData("time", readyTime.get())
+        .build();
+  }
+
+  public void onStartUp(
+      @Observes @Initialized(ApplicationScoped.class) Object init) {
+    readyTime = new AtomicLong(System.currentTimeMillis()); // <4>
+  }
+
+  /**
+   * Become ready after 5 seconds
+   *
+   * @return true if application ready
+   */
+  private boolean isStarted() {
+    return Duration.ofMillis(System.currentTimeMillis() - readyTime.get()).getSeconds() >= 8;
+  }
+}
+----
+<1> Include additional imports.
+<2> Annotation indicating that this is a startup health-check.
```

Comment:
General comment: hyphenated health-check

Reply:
@ljamen Usage question... Should we follow the "noun: no hyphen, adjective: hyphen" approach? There are places such as "health-check URL" (adjective) and the offending location you cited above (noun). I did some copying and pasting and might have propagated some correct or incorrect usages, and while I'm modifying the files I might as well make sure  they are consistently the way we want.

Reply:
BTW I also found a typo I have to fix so I'll be pushing more changes anyway.

Reply:
health check URL is the standard (no hyphen)